containers/python-3/.devcontainer/base.Dockerfile

```diff
@@ -0,0 +1,94 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+
+ARG VARIANT=3
+FROM python:${VARIANT}
+
+# If you would prefer to have multiple Python versions in your container,
+# replace the FROM statement above with the following:
+#
+# FROM ubuntu:bionic
+# ARG PYTHON_PACKAGES="python3.5 python3.6 python3.7 python3.8 python3 python3-pip"
+# RUN apt-get update && apt-get install --no-install-recommends -yq software-properties-common \
+#    && add-apt-repository ppa:deadsnakes/ppa && apt-get update \
+#    && apt-get install -yq --no-install-recommends ${PYTHON_PACKAGES} \
+#    && pip3 install --upgrade pip setuptools wheel
+
+# This Dockerfile adds a non-root user with sudo access. Use the "remoteUser"
+# property in devcontainer.json to use it. On Linux, the container user's GID/UIDs
+# will be updated to match your local UID/GID (when using the dockerFile property).
+# See https://aka.ms/vscode-remote/containers/non-root-user for details.
+ARG USERNAME=vscode
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+# Uncomment the following COPY line and the corresponding lines in the `RUN` command if you wish to
+# include your requirements in the image itself. Only do this if your requirements rarely change.
+# COPY requirements.txt /tmp/pip-tmp/
+
+# Set to false to skip installing zsh and Oh My ZSH!
+ARG INSTALL_ZSH="true"
+
+# Location and expected SHA for common setup script - SHA generated on release
+ARG COMMON_SCRIPT_SOURCE="https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/script-library/common-debian.sh"
+ARG COMMON_SCRIPT_SHA="dev-mode"
+
+# Avoid warnings by switching to noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Default set of utilities to install in a side virtual env
+ARG DEFAULT_UTILS="\
+    pylint \
+    flake8 \
+    autopep8 \
+    black \
+    pytest \
+    yapf \
+    mypy \
+    pydocstyle \
+    pycodestyle \
+    prospector \
+    pylama \
+    bandit \
+    virtualenv \
+    pipx"
```

Comment:
Should pipx be installed by pipx?

Reply:
Here's the question - would installing pipx globally affect dependencies too?  I was instead installing in an alternate location temporarily and then isolating it so it was available for further use.

Reply:
I guess if you're using pip to install pipx you're right that it should get installed on its own.

Reply:
Yeah I looked at using apt-get, but this ended up requiring installation venv and other 
OS dependencies as well all wired into the operating system version of python rather than the one from the image.

I could set `VIRTUALENV_PYTHON` to deal with the default there. Maybe that's enough? apt-get is certainly less gymnastics.

Reply:
Getting tied to the system Python isn't the worst thing in the world. I think it's a 🤷‍♂️ decision at this point. 😉